concordium-sdk/src/main/java/com/concordium/sdk/responses/BlockInfoV2.java

```diff
@@ -0,0 +1,29 @@
+package com.concordium.sdk.responses;
+
+import com.concordium.sdk.ClientV2;
+import com.concordium.sdk.transactions.Hash;
+import com.concordium.sdk.types.UInt64;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+
+/**
+ * Information about the Block. See {@link ClientV2#getBlocks()} & {@link ClientV2#getFinalizedBlocks()} ()}.
+ */
+@RequiredArgsConstructor
+@Builder
+@ToString
+@EqualsAndHashCode
+public class BlockInfoV2 {
```

Comment:
I suggest naming this otherwise as not to confuse it with `BlockInfo` as I just did. 

How about creating a class `BlockIdentifier` which holds these information, and then use that also in the existing `BlockInfo` ? (    private Integer blockHeight and private Hash blockHash; )

And note that the `BlockInfo` should still expose getBlockHeight() and getBlockHash() - which would then just call into the `BlockIdentifier`.

And then it should also be possible to pass a `BlockIdentifer` to the grpc in order to request a block. Maybe the `BlockIdentifier` should have a function which would return the response type you're using there.

Reply:
BlockInfoV2 has been renamed to BlockIdentifier.
BlockInfo is not derived from BlockIdentifier.

Regarding using BlockIdentifier as an input to fetch blocks. 
At various places where block and related information is being requested the API input parameter is a `Hash` (block hash). 
We can create an Overloaded method to also include `BlockIdentifier` since `BlockIdentifier` also has `blockHeight` which may or may not be available with Block`Hash` but that might not be necessary as Hash can be retrived from `BlockIdentifier`.

Reply:
Please see this comment. https://github.com/Concordium/concordium-java-sdk/pull/158#discussion_r1108183961 We should remain backwards compatible. 


Regarding the block identifier and using for querying calls. Let's just keep as is for now.